lib/scalapackfx.fpp

```diff
@@ -1975,7 +1981,86 @@ contains
     call infog2l(grow, gcol, desc, mygrid%nrow, mygrid%ncol, mygrid%myrow,&
         & mygrid%mycol, lrow, lcol, rsrc, csrc)
 
-  end subroutine scalafx_infog2l
+  end subroutine scalafx_infog2l_single
+
+  !> Maps global positions in a distributed matrix to local one.
+  !!
+  !! \param mygrid  BLACS descriptor.
+  !! \param desc  Descriptor of the distributed matrix.
+  !! \param grow  Global row indices.
+  !! \param gcol  Global column indices.
+  !! \param lrow  Local row indices on output.
+  !! \param lcol  Local column indices on output.
+  !! \param rsrc  Rows of the process owning the local matrix.
+  !! \param csrc  Columns of the process owning the local matrix.
+  !! \param calcAllIndices  Whether to calculate all lrow and lcol,
+  !!     even if the current process does not own them. (default: true)
```

Comment:
I am not sure, why this flag is needed. What relevance will `lrow` and `lcol` have on the processes, which do not own a given global position? Is there any scenario, where the local positions as calculated in this routine can be used on any processes but the one owning the global position? (Why not leave away this flag, and set the positions to `-1`, whenever the given global position is not local?)

Reply:
My intention was to not change the behavior of `infog2l`. If people change from the element-wise version to the array version, they might not expect a change in the result. That is why `calcAllIndices` is true by default.

Reply:
OK, that's a valid argument, so probably it is better to keep your version. I am still puzzled, what meaning those local positions actually have on the processes apart of the one which actually has the queried global position...